conans/client/cmd/new.py

```diff
@@ -236,13 +237,42 @@ def test(self):
 
 """
 
+def _render_template(text, name, version, package_name):
+    context = {'name': name, 
+               'version': version, 
+               'package_name': package_name,
+               'conan_version': client_version}
+    t = Template(text, keep_trailing_newline=True)
+    return t.render(**context)
+
+def _get_files_from_template_dir(template_dir, name, version, package_name):
+    if not os.path.isfile(os.path.join(template_dir, "conanfile.py")):
+        raise ConanException("Template is missing 'conanfile.py': {}".format(template_dir))
```

Comment:
IMO, we shouldn't add this check, the conanfile can take any other name, or maybe there is a `conanfile.txt`...

Reply:
I would really like to keep this type of check to prevent the following scenario: user makes a typo or template dir is malformed (e.g. `conan config install` did not go through), so the command `conan new --template=T` finishes successfully, but there is no files or worse a whole load of random files from the wrong location (the classic `cp -rf / path` mistake).

Reply:
If use enters a typo in `conan new --template=<typo>` it will fail because the folder/file doesn't exist. If there was a problem during the `conan config install` then, after this command runs successfully, you will realize about it in your working dir: missing files, a project that doesn't work... 

If you are worried about someone entering a relative path like `conan new --template=../../<typo>`, we can limit the search in the cache to folders inside the `<cache>/templates/cmd_new/` directory. It will work if it is found locally (`$(pwd)/../../<typo>`), but looking in the cache will raise.

Reply:
IIRC, `--template` accepts absolute paths as well. So, `conan new --template=/ path` will work (it will take huge time to complete though :D). This feature will also prevent users (which is a good thing, IMO) from (ab)using templates as a storage for random files.

Reply:
Ok, that was my opinion, but we can wait for others to review and add their two cents 👍

Reply:
Well, at the end of the day, you are the dev.
I.e. if you think that this check is not worth having then it's gotta go =)